spec/registry.md

```diff
@@ -0,0 +1,156 @@
+        <matchSignature>
+            <match values="windows linux macos android ios"/>
+        </matchSignature>
+    </function>
+
+    <pattern id="anyNumber" regex="-?[0-9]+(\.[0-9]+)"/>
+    <pattern id="positiveInteger" regex="[0-9]+"/>
+    <pattern id="currencyCode" regex="[A-Z]{3}"/>
+
+    <function name="number">
+        <description>
+            Format a number.
+            Match a numerical value against CLDR plural categories or against a number literal.
+        </description>
+
+        <matchSignature locales="en">
+            <input title="The number to match." pattern="anyNumber"/>
+            <option name="type" values="cardinal ordinal"/>
+            <option name="minimumIntegerDigits" pattern="positiveInteger"/>
+            <option name="minimumFractionDigits" pattern="positiveInteger"/>
+            <option name="maximumFractionDigits" pattern="positiveInteger"/>
+            <option name="minimumSignificantDigits" pattern="positiveInteger"/>
+            <option name="maximumSignificantDigits" pattern="positiveInteger"/>
+            <match values="one other"/>
+            <match pattern="anyNumber"/>
+        </matchSignature>
+
+        <formatSignature locales="en">
+            <input title="The number to format" pattern="anyNumber"/>
+            <option name="minimumIntegerDigits" pattern="positiveInteger"/>
+            <option name="minimumFractionDigits" pattern="positiveInteger"/>
+            <option name="maximumFractionDigits" pattern="positiveInteger"/>
+            <option name="minimumSignificantDigits" pattern="positiveInteger"/>
+            <option name="maximumSignificantDigits" pattern="positiveInteger"/>
+            <option name="style" readonly="true" values="decimal currency percent unit" default="decimal"/>
+            <option name="currency" readonly="true" pattern="currencyCode"/>
+        </formatSignature>
+    </function>
+</registry>
+```
+
+A localization engineer can then extend the registry by defining the following `customRegistry.xml` file.
+
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE registry SYSTEM "./registry.dtd">
+
+<registry>
+    <function name="noun">
+        <description>Handle the grammar of a noun.</description>
+        <formatSignature locales="en">
+            <input title="Noun id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="plural" values="one other"/>
+            <option name="case" values="nominative genitive" default="nominative"/>
+        </formatSignature>
+    </function>
+
+    <function name="adjective">
+        <description>Handle the grammar of an adjective.</description>
+        <formatSignature locales="en">
+            <input title="Adjective id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="plural" values="one other"/>
+            <option name="case" values="nominative genitive" default="nominative"/>
```

Comment:
I'm a little confused about why there's a `case` option for an adjective if this signature is only defined in locale `"en"`.

Reply:
English doesn't make it easy to build meaningful examples of grammatical features :) I'll try to come up with something better.

Reply:
Yeah, I know :) I was just thinking the example could be from a different language, maybe.